roseau/load_flow/models/branches.py

```diff
@@ -175,6 +189,9 @@ def _refresh_results(self) -> None:
         if self._fetch_results:
             self._res_currents = self._cy_element.get_side_currents(self._n1, self._n2)
             self._res_potentials = self._cy_element.get_side_potentials(self._n1, self._n2)
+            if self._has_ground:
+                all_potentials = self._cy_element.get_potentials(self._n)
+                self._res_ground_potential = all_potentials.item(-1)
```

Comment:
This is not the fastest way to get results as we are getting twice the potentials. If speed is problematic we can add a C++ methods to get all the 3 potentials of a shunt line.  (this method is not particularly slow, but it can typically get called a lot of times)

Reply:
Yeah it would be great if we can ask for the current/potential results of a certain port in C++. Say something like `get_port_potential(i)` where `i` is the index of the port. This could exist on the `Element` class, no need to special case shunt lines. It could be useful for other elements as well that only have one useful port like grounds, prefs and ground connections. The entire RLFS could also use the more efficient methods.

Reply:
Yes good idea, I can add it in my next PR in `engine`

Reply:
This now uses `self._cy_element.get_port_potential`